app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -156,7 +160,7 @@ open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus: Bus,
   }
 
   private fun onRestoreViewStateSearchResults() {
-    mainViewController?.hideSettingsBtn()
+    // Do nothing.
```

Comment:
Per our discussion in slack, this is because we need the `when` statement to be exhaustive. Let's try out moving these into just anonymous blocks that get executed at the actual `when` statement rather than separate functions. Should make the code more concise and readable.